### PR TITLE
Move FP8 to sglang

### PR DESCRIPTION
**Pull request by HaiShaw**
## Motivation

Move FP8 layers definition to SGLang

## Modifications

As it is.
Kernels come next.

## Checklist

- [+] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [+] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [+] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

File: python/sglang/srt/layers/quantization/fp8.py [hunk 1]
```diff
@@ -0,0 +1,564 @@
+# Adapted from https://github.com/vllm-project/vllm/blob/v0.6.4.post1/vllm/model_executor/layers/quantization/fp8.py
+
+from typing import Any, Callable, Dict, List, Optional
+
+import torch
+import vllm.envs as envs
+from torch.nn import Module
+from torch.nn.parameter import Parameter
+from vllm import _custom_ops as ops
+from vllm.logger import init_logger
```
Comment (zhyncs) [inline python/sglang/srt/layers/quantization/fp8.py]: remove this

File: python/sglang/srt/layers/quantization/fp8.py [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+from vllm.model_executor.layers.quantization.utils.quant_utils import is_layer_skipped
+from vllm.model_executor.layers.quantization.utils.w8a8_utils import (
+    all_close_1d,
+    apply_fp8_linear,
+    convert_to_channelwise,
+    cutlass_fp8_supported,
+    per_tensor_dequantize,
+    requantize_with_max_scale,
+)
+from vllm.model_executor.parameter import ModelWeightParameter, PerTensorScaleParameter
```
Comment (zhyncs) [inline python/sglang/srt/layers/quantization/fp8.py:25]: remove this

File: python/sglang/srt/layers/quantization/fp8.py [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+from vllm.model_executor.utils import set_weight_attrs
```
Comment (zhyncs) [inline python/sglang/srt/layers/quantization/fp8.py]: remove this

File: python/sglang/srt/layers/quantization/fp8.py [hunk 4]
```diff
@@ -0,0 +1,564 @@
...
+from vllm.platforms import current_platform
```
Comment (zhyncs) [inline python/sglang/srt/layers/quantization/fp8.py]: remove this

File: python/sglang/srt/layers/quantization/fp8.py [hunk 5]
```diff
@@ -0,0 +1,564 @@
...
+from vllm.utils import print_warning_once
```
Comment (zhyncs) [inline python/sglang/srt/layers/quantization/fp8.py]: remove this

File: python/sglang/srt/layers/quantization/fp8.py [hunk 6]
```diff
@@ -0,0 +1,564 @@
+# Adapted from https://github.com/vllm-project/vllm/blob/v0.6.4.post1/vllm/model_executor/layers/quantization/fp8.py
+
+from typing import Any, Callable, Dict, List, Optional
+
+import torch
+import vllm.envs as envs
```
Comment (zhyncs) [inline python/sglang/srt/layers/quantization/fp8.py]: remove this

**Review by zhyncs — COMMENTED** [review]
Except for vllm.model_executor.layers.quantization, LinearBase, and _custom_ops, everything else needs to be removed. Thanks!

File: python/sglang/srt/layers/quantization/fp8.py (hunk 2 above)
Comment (HaiShaw) [reply python/sglang/srt/layers/quantization/fp8.py:25]: This is still in use, will decouple and migrate later.

**Comment by zhyncs** [comment]
move to https://github.com/sgl-project/sglang/pull/2370
All credit goes to @HaiShaw Thanks!

**Pull request closed without merging**